client/components/common/BaseForm.vue

```diff
@@ -95,6 +103,7 @@ export default {
     this.fields.forEach(f => {
       const validators = [
         !f.optional && (v => v ? '' : 'required field'),
+        !f.optional && this.$helpers.validators.nonEmpty,
```

Comment:
I'd rather not add this since we already added in all the other specific places

Reply:
that would be reasonable as well. I'm going offline now, but feel free to do it that way across the board if you prefer.  My only objection is to the duplication.